packages/@aws-cdk/aws-eks/lib/cluster-resource-handler/cluster.ts

```diff
@@ -22,8 +22,24 @@ export class ClusterResourceHandler extends ResourceHandler {
   constructor(eks: EksClient, event: ResourceEvent) {
     super(eks, event);
 
-    this.newProps = parseProps(this.event.ResourceProperties);
-    this.oldProps = event.RequestType === 'Update' ? parseProps(event.OldResourceProperties) : { };
+    function patchEndpointAccess(props: any) {
```

Comment:
Move this function outside the ctor

Reply:
Not sure I understand why as this function is really specific and should only be used in this context.
But in any case I merged it with `parseProps`.